content/en/docs/guides/server-graceful-stop.md

```diff
@@ -0,0 +1,88 @@
+---
+title: Gracefully stopping a server
+description: >-
+  Explains how to gracefully stop/shutdown a gRPC server.
+---
+
+### Overview
+
+gRPC servers often need to shut down gracefully, ensuring that in-flight RPCs
+are completed within a reasonable timeframe and new RPCs are no longer
+accepted. The "Graceful shutdown function" facilitates this process, allowing
+the server to transition smoothly without abruptly terminating active
+connections.
+
+When the "Graceful shutdown function" is called, the server immediately stops
+accepting new RPCs. In-flight RPCs are allowed to continue until they complete
+or a specified deadline is reached. Once all active RPCs finish or the deadline
+expires, the server shuts down completely.
+
+### How to do Graceful Server Shutdown
+
+The exact implementation of the "Graceful shutdown function" varies depending on
+the programming language you are using. However, the general pattern
+involves:
+
+- Initiating the graceful shutdown process by calling "Graceful shutdown
+  Function" on your gRPC server object. This function blocks until all
+  currently running RPCs complete. This ensures that in-flight requests are
+  allowed to finish processing.
+- Specify a timeout period to limit the time allowed for in-progress RPCs to
+  finish. It's crucial to separately call the "Forceful shutdown function" on
+  the server object using a timer mechanism (depending on your language) to
+  trigger a forceful shutdown after a predefined duration. This
+  acts as a safety net, ensuring that the server eventually shuts down even if
+  some in-flight RPCs don't complete within a reasonable timeframe. This
+  prevents indefinite blocking.
+
+The following shows the sequence of events that occur, when a server's graceful
+stop is invoked, in-flight RPCs continue to prorcess, but new RPCs are
+rejected. If some in-flight RPCs are not finished in time, server is forcefully
+shutdown.
+```mermaid
+sequenceDiagram
+Client->>Server: New RPC Request 1
+Client->>Server: New RPC Request 2
+Server->>Client: Processing RPC 1
+Server->>Client: Processing RPC 2
+Server-->>Server: Graceful Stop Invoked
+Server->>Client: Continues Processing In-Flight RPCs
+Client->>Server: New RPC Request 3 (Rejected)
+alt RPCs complete within timeout
+    Server->>Client: Completes RPC 1
+    Server->>Client: Completes RPC 2
+    Server-->>Server: Shutdown Complete
+    Client->>Server: New RPC Request 4 (Rejected)
```

Comment:
Once a connection has been closed, RPCs aren't technically "rejected" anymore.  They literally can't even be attempted.

Also should we discuss that the client will detect the server shutting down and go find other servers to send its traffic to, if at all possible?

Reply:
Removed the rejected part and added the client will find another server if applicable